samples/attestation/common/crypto.cpp

```diff
@@ -2,96 +2,90 @@
 // Licensed under the MIT License.
 
 #include "crypto.h"
-#include <mbedtls/pk.h>
-#include <mbedtls/rsa.h>
 #include <openenclave/enclave.h>
+#include <openssl/bio.h>
+#include <openssl/crypto.h>
+#include <openssl/evp.h>
+#include <openssl/pem.h>
+#include <openssl/rsa.h>
 #include <stdlib.h>
 #include <string.h>
 
 Crypto::Crypto()
 {
-    m_initialized = init_mbedtls();
+    m_initialized = init_openssl3();
 }
 
 Crypto::~Crypto()
 {
-    cleanup_mbedtls();
+    cleanup_openssl3();
 }
 
 /**
- * init_mbedtls initializes the crypto module.
- * mbedtls initialization. Please refer to mbedtls documentation for detailed
+ * init_openssl3 initializes the crypto module.
+ * openssl 3 initialization. Please refer to openssl documentation for detailed
  * information about the functions used.
  */
-bool Crypto::init_mbedtls(void)
+bool Crypto::init_openssl3(void)
 {
     bool ret = false;
-    int res = -1;
+    BIO* mem = nullptr;
+    size_t numbytes = 0;
+    EVP_PKEY_CTX* ctx = nullptr;
+    char* bio_ptr = nullptr;
 
-    mbedtls_ctr_drbg_init(&m_ctr_drbg_contex);
-    mbedtls_entropy_init(&m_entropy_context);
-    mbedtls_pk_init(&m_pk_context);
-
-    // Initialize entropy.
-    res = mbedtls_ctr_drbg_seed(
-        &m_ctr_drbg_contex,
-        mbedtls_entropy_func,
-        &m_entropy_context,
-        nullptr,
-        0);
-    if (res != 0)
+    if (!(ctx = EVP_PKEY_CTX_new_from_name(NULL, "RSA", NULL)))
     {
-        TRACE_ENCLAVE("mbedtls_ctr_drbg_seed failed.");
+        TRACE_ENCLAVE("EVP_PKEY_CTX_new_from_name failed!\n");
         goto exit;
     }
-
-    // Initialize RSA context.
-    res = mbedtls_pk_setup(
-        &m_pk_context, mbedtls_pk_info_from_type(MBEDTLS_PK_RSA));
-    if (res != 0)
+    if (!EVP_PKEY_keygen_init(ctx))
     {
-        TRACE_ENCLAVE("mbedtls_pk_setup failed (%d).", res);
+        TRACE_ENCLAVE("EVP_PKEY_keygen_init failed!\n");
         goto exit;
     }
-
-    // Generate an ephemeral 2048-bit RSA key pair with
-    // exponent 65537 for the enclave.
-    res = mbedtls_rsa_gen_key(
-        mbedtls_pk_rsa(m_pk_context),
-        mbedtls_ctr_drbg_random,
-        &m_ctr_drbg_contex,
-        2048,
-        65537);
-    if (res != 0)
+    if (!EVP_PKEY_generate(ctx, &rsa_pkey))
```

Comment:
Let's leave a comment here saying that OpenSSL generates RSA key pair using 2048-bit and exponent 65537 by default.
https://docs.openssl.org/3.0/man3/EVP_PKEY_CTX_ctrl/#rsa-parameters